### ISSUE TITLE
Winner Offer

### ISSUE THREAD

**Issue by Braandn** [TheAnthonyNL/SteamBot#12]
Everytime when the winner is selected, the bot throw out errors about the Steam API KEY

Log:
http://hastebin.com/wageqorifa.vbs

I've updated all API Keys in Bot and website but I don't know how to fix it...
The winner offer will be sent but the profit offer not!

**Comment by Xeroner** [comment]
For this moment nobody know what is this API errors...

**Comment by GreYzZ** [comment]
I get the same errors

**Comment by TheAnthonyNL** [comment]
Well i find it strange how you guys get errors while it works for me..

**Comment by GreYzZ** [comment]
@TheAnthonyNL 
yes I think it's because there is a problem with this exec 0 withdraw not working.
So the bot cant send me a offer because he got no tradelink from me

**Comment by TheAnthonyNL** [comment]
You need to be friends with bot ofc...

**Comment by MilyGosc** [comment]
@TheAnthonyNL 
So can you answer on few questions?
1. What is your system for bot?
2. If it's linux, which verison of mono you have?
3. You are in Visual Studio building project ExampleBot or solution?

Thanks in advance

**Comment by TheAnthonyNL** [comment]
Just updated the bot last night.

**Comment by linkzy** [comment]
Hello!  @TheAnthonyNL 

First i would like to thank you for all your hard work on the bot. If it weren't for you i would not be able start my site.

This is an old issue, but I just found out why there are so many errors on the winner offer. After 3000 pots on my site, over 500 had to be paid manually because of this error.

What happens is this: when an user makes a depoist, the bot saves the "classId" and the "instaceId" from every item, so later it can use this values to find "assetId" to add this item to the winner offer.

The problem is... depending on the item, when the bot accepts the trade offer, the "classId" and the "instanceId" from an item may change... https://www.reddit.com/r/SteamBot/comments/394o9d/understanding_item_ids/

So basically, the bot looks for the old "classId" and "instanceId" on the inventory... it doesnt find it... then it adds an item with the assetId = 0 to the winner offer. When the offer is sent, steam returns an error.

Since i couldnt find a way to get the new ids after the offer is accepted, i will try to change my bot to find the new ids based on the name of the item and then find "assetID".

This will cause another problem for me... if an user takes too long to accept an offer... the item may be sent to another winner with similar itens on the pot. So i will have to store the "assetId" from the itens i already sent... so the bot can check and avoid sending the same iten to two different winners.

Sorry for the long post and for my bad english. If anyone has a better idea to solve this problem, please share =)

Thanks again @TheAnthonyNL for the bot!

**Comment by mx0c** [comment]
@linkzy may i ask you whats the domain of your site ? And are you interested to cooperate with me to code a 1v1? kust pm me or add me on steam: http://steamcommunity.com/id/mx0c/

**Comment by linkzy** [comment]
@mx0c I had to close my site =(

I decided to create a new site and new bot based on the csgowinbig... but the old site issues were taking so many time from me that i had to close the old site to be able to work on the new one.

This is what it used to be:
http://loteriacs.com/index3.html

This is what im working on:
http://loteriacs.com/beta/
(The bot is offline and if you login part of the content is static)

Im going to add you on steam =)
